src/ast/passes/codegen_llvm.cpp

```diff
@@ -1291,11 +1291,26 @@ void CodegenLLVM::visit(Call &call)
     auto &arg = *call.vargs.at(0);
     auto &map = static_cast<Map &>(arg);
 
-    if (!map_len_func_)
-      map_len_func_ = createMapLenCallback();
+    // Some map types used in bpftrace (BPF_MAP_TYPE_(PERCPU_)ARRAY) do not
+    // implement per-cpu counters and bpf_map_sum_elem_count would always return
+    // 0 for them. In our case, those maps typically have a single element so we
+    // can return 1 straight away.
+    // For the rest, use bpf_map_sum_elem_count if available and map supports
+    // it, otherwise fall back to bpf_for_each_map_elem with a custom callback.
+    if (map_has_single_elem(map.type, map.key_type)) {
+      expr_ = b_.getInt64(1);
+    } else if (bpftrace_.feature_->has_kernel_func("bpf_map_sum_elem_count") &&
+               !is_array_map(map.type, map.key_type)) {
```

Comment:
Don't we statically know size of array map? Should be possible to just hard code the result into bytecode, right? So another branch like single elem case above

Reply:
We do but this case checks if the map is **not** an array.

Reply:
Right, cuz array map is statically sized. I was suggesting we could have above a:

```
else if (is_array_map(...) {
  expr_ = b_.getInt64(map.type.GetNumElements());
}
```

then the `!is_array_map(..)` check in the has_kernel_func() branch could be dropped as well.